bin/client/src/interop/transition.rs

```diff
@@ -31,31 +31,42 @@ pub(crate) async fn sub_transition<P, H>(
         >,
     >,
     boot: BootInfo,
-    pre: PreState,
 ) -> Result<(), FaultProofProgramError>
 where
     P: PreimageOracleClient + Send + Sync + Debug + Clone,
     H: HintWriterClient + Send + Sync + Debug + Clone,
 {
     // Check if we can short-circuit the transition, if we are within padding.
-    if let PreState::TransitionState(ref transition_state) = pre {
+    if let PreState::TransitionState(ref transition_state) = boot.agreed_pre_state {
         if transition_state.step >= transition_state.pre_state.output_roots.len() as u64 {
             info!(
                 target: "interop_client",
                 "No derivation/execution required, transition state is already saturated."
             );
 
-            return transition_and_check(pre, None, boot.claimed_post_state, None);
+            return transition_and_check(boot.agreed_pre_state, None, boot.claimed_post_state, None);
         }
     }
 
     // Fetch the L2 block hash of the current safe head.
-    let safe_head_hash = fetch_l2_safe_head_hash(oracle.as_ref(), &pre).await?;
+    let safe_head_hash = fetch_l2_safe_head_hash(oracle.as_ref(), &boot.agreed_pre_state).await?;
+
+    // Determine the active L2 chain ID and the fetch rollup configuration.
+    let active_l2_chain_id = boot
+        .agreed_pre_state
+        .active_l2_chain_id()
+        .ok_or(FaultProofProgramError::StateTransitionFailed)?;
+    let rollup_config = boot
+        .rollup_configs
+        .get(&active_l2_chain_id)
+        .cloned()
+        .map(Arc::new)
+        .ok_or(FaultProofProgramError::StateTransitionFailed)?;
```

Comment:
Seems like we could pull this into a function on the `boot` itself.

e.g.
```rust
impl BootInfo {
    /// Returns the [RollupConfig] for the active L2 chain of the boot info.
    pub fn active_rollup_config(&self) -> Result<RollupConfig, BootInfoError> {
        let active_l2_chain_id = self.agreed_pre_state.active_l2_chain_id().ok_or(BootInfoError::MissingActiveChainId)?;
        self.rollup_configs.get(&active_l2_chain_id).cloned().ok_or(BootInfoError::MissingRollupConfig(active_l2_chain_id)
    }
}
```

Reply:
Yeah will do :)